bigtools/src/utils/misc.rs

```diff
@@ -51,17 +53,21 @@ pub fn stats_for_bed_item<R: BBIFileRead>(
 
     let mut bases = 0;
     let mut sum = 0.0;
+    let mut min = f64::MAX;
+    let mut max = f64::MIN;
     for val in interval {
         let num_bases = val.end - val.start;
         bases += num_bases;
         sum += f64::from(num_bases) * f64::from(val.value);
+        min = min.min(f64::from(val.value));
+        max = max.max(f64::from(val.value));
     }
     let size = end - start;
     let mean0 = sum / f64::from(size);
-    let mean = if bases == 0 {
-        0.0
-    } else {
-        sum / f64::from(bases)
+    let (mean, min, max) = match (bases == 0, end - start == bases) {
```

Comment:
I think I might be missing the point of this `end - start == bases` check. Oh, I see.

I think we probably want to treat uncovered data as nan, not 0 for min/max. So this check isn't needed.

Reply:
I mimicked the values `bigWigAverageOverBed -minMax`  returns for regions for which there are no entries in the bigWig file.

Reply:
Does UCSC's bigWigAverageOverBed treat uncovered as 0? Looking [here](https://github.com/ucscGenomeBrowser/kent/blob/ec8e394fe288b94187c9258a0aa02a396d7c3ab0/src/utils/bigWigAverageOverBed/bigWigAverageOverBed.c#L71) I don't see that.

Reply:
To me it at least looks like it:

```bash
# Create BED file with 2 regions: one without any coverage in the BED file and one with (at least partial coverage):
$ printf 'chr1\t3296410\t3296910\tno_coverage\nchr1\t3296410\t3297910\tsome_coverage\n' > overlap.bed

$ cat overlap.bed
chr1	3296410	3296910	no_coverage
chr1	3296410	3297910	some_coverage

# With bigWigAverageOverBed of Kent tools:
./bigWigAverageOverBed -minMax test.bw overlap.bed /dev/stdout | head
no_coverage	500	0	0	0	0	0	0
some_coverage	1500	376	6.40354	0.00426903	0.0170307	0.0149267	0.0298533


# First BED entry does not have coverage in the bigWig file: chr1	3296410	3296910
❯ bigtools bigwigtobedgraph test.bw /dev/stdout overlap.bed | head
chr1	3296983	3297037	0.014926667
chr1	3297407	3297549	0.014926667
chr1	3297549	3297574	0.029853335
chr1	3297574	3297606	0.014926667
chr1	3297787	3297882	0.014926667
chr1	3297882	3297910	0.029853335
```

Reply:
Found it: https://github.com/ucscGenomeBrowser/kent/blob/ec8e394fe288b94187c9258a0aa02a396d7c3ab0/src/utils/bigWigAverageOverBed/bigWigAverageOverBed.c#L202-L213

Reply:
It might be good to have a `min0` and `max0` similar to `mean0`, besides `min` and `max`.

Reply:
So, this is different, right? If there is *no coverage* indeed we should return 0 (and there is already code for this). But, if there is only *some* coverage, the minimum should be the minimum for covered bases, not 0.

Just so it's clear, it seems like this check is basically checking that the covered bases (bases) is the whole interval (end - start); and if not, setting the minimum to 0.

Reply:
I think I got tricked due to a bug in Kent tools bigWigAverageOverBed
```
# Processing a bunch of regions with Kent tools bigWigAverageOverBed:
$ ./bigWigAverageOverBed -minMax test.bw consensus_peaks_bicnn.bed consensus_peaks.ucsc.bwaob
processing chromosomes....................

# Looking at a specific region:
$ grep chr1:3265027-3265527 consensus_peaks.ucsc.bwaob
chr1:3265027-3265527	500	390	6.31398	0.012628	0.0161897	0	0.0298533

# Creating a new BED file with only that region:
$ grep chr1:3265027-3265527 consensus_peaks_bicnn.bed > consensus_peaks_bicnn.chr1:3265027-3265527.bed

$ cat consensus_peaks_bicnn.chr1:3265027-3265527.bed
chr1	3265027	3265527	chr1:3265027-3265527

# Processing only this region with Kent tools bigWigAverageOverBed:
$ ./bigWigAverageOverBed -minMax pybigtools/Astro.bw consensus_peaks.chr1\:3265027-3265527.bed /dev/stdout
chr1:3265027-3265527	500	390	6.31398	0.012628	0.0161897	0.0149267	0.0298533

==> different minimum
```

Reply:
No reply yet from UCSC about the issue: https://groups.google.com/a/soe.ucsc.edu/g/genome/c/fh9nNK52Qrw

Reply:
Just checking, is `test.bw` the same as `pybigtools/Astro.bw`?

I'm not quite sure what you expect the issue is?

The minimum in the output is `0.0149267`, which is the minimum in the bigwig file (rounded). Of course, `3265417-3265527` is uncovered, which verifies that it is not treated as `0`.

Reply:
Oh, I see what you mean. You get a different minimum when you subset to only a single peak.

Reply:
That's really strange, but almost certainly a bug on their end.